src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueue.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+
+namespace Enable.Extensions.Queuing.AzureServiceBus.Internal
+{
+    internal class AzureServiceBusQueue : IDisposable
+    {
+        private int _referenceCount = 1;
+        private bool _disposed = false;
+
+        public AzureServiceBusQueue(string connectionString, string queueName)
+        {
+            Receiver = new MessageReceiver(
+                connectionString,
+                queueName,
+                ReceiveMode.PeekLock);
+
+            Sender = new MessageSender(connectionString, queueName);
+        }
+
+        public IMessageSender Sender { get; }
+        public IMessageReceiver Receiver { get; }
+
+        public int AddReference()
+        {
+            return Interlocked.Increment(ref _referenceCount);
+        }
+
+        public int RemoveReference()
+        {
+            return Interlocked.Decrement(ref _referenceCount);
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
```

Comment:
All other implementations of `Dispose` in this project contain a `GC.SuppressFinalize(this);`. Let's add that here too.

Reply:
Done.